pkg/standalone-utils/contracts/LPOracleFactoryBase.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { AggregatorV3Interface } from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { ILPOracleFactoryBase } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/ILPOracleFactoryBase.sol";
+import { ILPOracleBase } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/ILPOracleBase.sol";
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
+
+abstract contract LPOracleFactoryBase is ILPOracleFactoryBase, SingletonAuthentication {
+    uint256 internal _oracleVersion;
```

Comment:
Is there a specific reason to use a numerical version? Should we just use Version?

Reply:
+1, let's just shift to `Version`

Reply:
So, AggregatorV3Intervace has a `version` function that requires uint256, while our Version library is of type `string`. To be compatible with chainlink's interace, I'd recommend leaving it as is and don't adopting the Version library.

Reply:
fair point, agreed.